logging/src/main/java/org/jboss/as/logging/LoggingResource.java

```diff
@@ -290,7 +319,10 @@ public FileVisitResult visitFile(final Path file, final BasicFileAttributes attr
                     // This should always exist, but better to be safe
                     if (handlerModel.hasDefined(CommonAttributes.FILE.getName())) {
                         final ModelNode fileModel = handlerModel.get(CommonAttributes.FILE.getName());
-                        if (fileModel.hasDefined(PathResourceDefinition.PATH.getName())) {
+                        // Only allow from the jboss.server.log.dir
+                        if (fileModel.has(PathResourceDefinition.RELATIVE_TO.getName())
```

Comment:
hasDefined would be better. A proper resource model will return true for "has" for all attributes as the add handler will populate everything either with a normal value or undefined.